SimG4CMS/Muon/src/MuonSensitiveDetector.cc

```diff
@@ -336,3 +353,35 @@ Local3DPoint MuonSensitiveDetector::FinalStepPositionVsParent(const G4Step* curr
 
   return ConvertToLocal3DPoint(localCoordinates);
 }
+
+bool MuonSensitiveDetector::acceptHit(uint32_t id) {
+  bool flag(true);
+  if (haveDemo_) {
+    int subdet = DetId(id).subdetId();
+    if (subdet == MuonSubdetId::GEM) {
+      if (GEMDetId(id).station() == 2)
+        flag = demoGEM_;
+    } else if (subdet == MuonSubdetId::RPC) {
+      if ((RPCDetId(id).region() != 0) && (RPCDetId(id).ring() == 1) && (RPCDetId(id).station() > 2))
+        flag = demoRPC_;
+    }
+  }
+  if (id == 0)
```

Comment:
Shouldn't this be checked at the very beginning of the method, and simply return `false` if `id == 0` ?
(Possibly with a debug line attached, if you want so)

Reply:
Taken care of